src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java

```diff
@@ -0,0 +1,167 @@
+                    .add("you cannot remove variations to Oredict chisel groups {}", groupName)
+                    .add("instead, edit the oredict via `oredict.remove('{}', {})`", groupName, IngredientHelper.asGroovyCode(item, false))
+                    .error()
+                    .post();
+        }
+    }
+
+    public void setSound(String group, SoundEvent sound) {
+        setSound(CarvingUtils.getChiselRegistry().getGroup(group), sound);
+    }
+
+    public void setSound(ICarvingGroup group, SoundEvent sound) {
+        CarvingUtils.getChiselRegistry().setVariationSound(group.getName(), sound);
+        this.sounds.add(Pair.of(group.getName(), group.getSound()));
+    }
+
+    public void addGroup(String groupName) {
+        if (CarvingUtils.getChiselRegistry().getSortedGroupNames().contains(groupName)) {
+            GroovyLog.msg("Error adding Chisel Carving group")
+                    .add("found a duplicate Chisel Carving group with name {}", groupName)
+                    .error()
+                    .post();
+            return;
+        }
+        CarvingUtils.getChiselRegistry().addGroup(CarvingUtils.getDefaultGroupFor(groupName));
+        this.groupScripted.add(groupName);
+    }
+
+    public void removeGroup(String groupName) {
+        if (!CarvingUtils.getChiselRegistry().getSortedGroupNames().contains(groupName)) {
+            GroovyLog.msg("Error removing Chisel Carving group")
+                    .add("could not find Chisel Carving group with name {}", groupName)
+                    .error()
+                    .post();
+            return;
+        }
+        CarvingUtils.getChiselRegistry().removeGroup(groupName);
+        this.groupBackup.add(groupName);
+    }
+
+    public void removeAll() {
+        CarvingUtils.getChiselRegistry().getSortedGroupNames().forEach(name -> {
+            CarvingUtils.getChiselRegistry().removeGroup(name);
+            this.groupBackup.add(name);
+        });
+    }
+
+
+    public static class CarvingGroup {
+
+        ICarvingGroup group;
+
+        public CarvingGroup(String group) {
+            if (!CarvingUtils.getChiselRegistry().getSortedGroupNames().contains(group)) ModSupport.CHISEL.get().carving.addGroup(group);
+            this.group = CarvingUtils.getChiselRegistry().getGroup(group);
+        }
+
+        public CarvingGroup sound(SoundEvent sound) {
+            ModSupport.CHISEL.get().carving.setSound(group, sound);
+            return this;
+        }
+
+        public CarvingGroup add(ItemStack item) {
+            ModSupport.CHISEL.get().carving.addVariation(this.group.getName(), item);
+            return this;
+        }
+
+        public CarvingGroup add(ItemStack... items) {
+            for (ItemStack item : items) {
+                add(item);
+            }
+            return this;
+        }
+
+        public CarvingGroup add(Collection<ItemStack> items) {
+            for (ItemStack item : items) {
```

Comment:
`items.forEach(this::add)` can be used here

Reply:
its whats used in `AbstractRecipeBuilder`, so i would prefer not to change this